numpy/core/__init__.py

```diff
@@ -75,8 +75,10 @@
 from . import fromnumeric
 from .fromnumeric import *
 from . import defchararray as char
+from . import records
 from . import records as rec
 from .records import record, recarray, format_parser
+from . import memmap
 from .memmap import *
```

Comment:
There's a `__getattr__` in `core/__init__.py`, so it seems like a bug in `importlib` that this is necessary to begin with.

Reply:
But you add `memmap` to `__all__` later.  So it needs to be in the namespace. The standard import automagically adds it...  which is a little odd.  But the lazy import doesn't. And that's where the trouble comes from. This just mimics what the standard import does.

Reply:
Not quite I think (my `__getattr__` comment may be confusing though). What I meant above is: what is in the namespace above is the _class_ `memmap` in `core/memmap.py`; this is what is added to `__all__`. There is no point in adding the _module_ `memmap` here, it is not needed and is immediately overwritten in the `from .memmap import *`.

Reply:
Reusing the same name is also considered a bug/anti-pattern by Mypy: https://github.com/python/mypy/issues/1174. Or maybe "not supported", this has tripped me up several times already. We should probably move the `memmap.py` file to `_memmap.py`, this aliasing is always confusing.

Reply:
Got it!  Thanks for clueing me in to that. I'll take that out...  maybe with a comment.